app/Models/Camp.php

```diff
@@ -193,4 +195,153 @@ public static function getAllParentCamp($topicNum, $filter, $asOfDate = null)
         ->whereRaw('go_live_time in (select max(go_live_time) from camp where topic_num=' . $topicNum . ' and objector_nick_id is null and go_live_time < '.$asOfDate.' group by camp_num)')
         ->orderBy('submit_time', 'desc')->orderBy('camp_name','desc')->groupBy('camp_num')->get();
     }
+
+    public static function getAllChildCamps($camp) {
+        $camparray = [];
+        if ($camp) {
+            $key = $camp->topic_num . '-' . $camp->camp_num . '-' . $camp->parent_camp_num;
+            $key1 = $camp->topic_num . '-' . $camp->parent_camp_num . '-' . $camp->camp_num;
+           if (in_array($key, Camp::$chilcampArray) || in_array($key1, Camp::$chilcampArray)) {
+                return [];
+            }
+            Camp::$chilcampArray[] = $key;
+            Camp::$chilcampArray[] = $key1;
+            $camparray[] = $camp->camp_num;
+            $childCamps = Camp::where('topic_num', $camp->topic_num)->where('parent_camp_num', $camp->camp_num)->where('go_live_time', '<=', time())->groupBy('camp_num')->latest('submit_time')->get();
+            foreach ($childCamps as $child) {
+                $latestParent = Camp::where('topic_num', $child->topic_num)
+                ->where('camp_num', $child->camp_num)->where('go_live_time', '<=', time())->latest('submit_time')->first();
+                if($latestParent->parent_camp_num == $camp->camp_num ){ 
+                    $camparray = array_merge($camparray, self::getAllChildCamps($child)); 
+
+                }
+                
+            }
+        }
+
+        return $camparray;
+    }
+
+    public static function getCampSubscribers($topic_num, $camp_num = 1){
+
+        $users_data = [];
+        $users = CampSubscription::select('user_id')->where('topic_num', '=', $topic_num)
+            ->whereIn('camp_num', [0, $camp_num])
+            ->get();
+        if (count($users)) {
+            foreach ($users as $user) {
+                array_push($users_data, $user->user_id);
+            }
+        }
+        if ($camp_num) {
+            $filter['topicNum'] = $topic_num;
+            $filter['asOf'] = '';
+            $filter['campNum'] = $camp_num;
+            $onecamp = self::getLiveCamp($filter);
+        } else {
+            $onecamp = self::getLiveCampFromTopic($topic_num, ['nofilter' => true]);
+        }
+        $childCampData = [];
+        if (isset($onecamp) && isset($onecamp->camp_name)) {
+            if ($camp_num) {
+                $childCampData = $onecamp->campChild($topic_num, $camp_num);
+            } else {
+                $childCampData = self::campChildFromTopic($topic_num);
+            }
+        }
+        $child_camps = [];
+        if (count($childCampData) > 0) {
+            foreach ($childCampData as $key => $child) {
+                $child_camps[$key] = $child->camp_num;
+            }
+        }
+        if (count($child_camps) > 0) {
+
+            $usersData = CampSubscription::select('user_id')->where('topic_num', '=', $topic_num)
```

Comment:
Can we combine the query above $users and this CampSubscription into one? I think we can. We just need to think a more on it.